tokenizers/src/pre_tokenizers/split.rs

```diff
@@ -0,0 +1,145 @@
+            invert: false,
+        }
+    }
+}
+
+impl Split {
+    pub fn new(re: Regex, behavior: SplitDelimiterBehavior, invert: bool) -> Self {
+        Split {
+            re,
+            behavior,
+            invert
+        }
+    }
+}
+
+impl PreTokenizer for Split {
+    fn pre_tokenize(&self, pretokenized: &mut PreTokenizedString) -> Result<()> {
+        use SplitDelimiterBehavior::*;
+        match self.behavior {
+            Removed => {
+                if self.invert { 
+                    pretokenized.split(|_, normalized| {
+                        normalized.split(Invert(&self.re),Removed)
+                    })
+                } else {
+                    pretokenized.split(|_, normalized| {
+                        normalized.split(&self.re, Removed)
+                    })
+                }
+            },
+            Isolated => {
+                if self.invert { 
+                    pretokenized.split(|_, normalized| {
+                        normalized.split(Invert(&self.re), Isolated)
+                    })
+                } else {
+                    pretokenized.split(|_, normalized| {
+                        normalized.split(&self.re, Isolated)
+                    })
+                }  
+            },
+            MergedWithPrevious => {
+                if self.invert { 
+                    pretokenized.split(|_, normalized| {
+                        normalized.split(Invert(&self.re), MergedWithPrevious)
+                    })
+                } else {
+                    pretokenized.split(|_, normalized| {
+                        normalized.split(&self.re, MergedWithPrevious)
+                    })
+                }  
+            },
+            MergedWithNext => {
+                if self.invert { 
+                    pretokenized.split(|_, normalized| {
+                        normalized.split(Invert(&self.re), MergedWithNext)
+                    })
+                } else {
+                    pretokenized.split(|_, normalized| {
+                        normalized.split(&self.re, MergedWithNext)
+                    })
+                }  
+            },
+            Contiguous => {
+                if self.invert { 
+                    pretokenized.split(|_, normalized| {
+                        normalized.split(Invert(&self.re), Contiguous)
+                    })
+                } else {
+                    pretokenized.split(|_, normalized| {
+                        normalized.split(&self.re, Contiguous)
+                    })
+                }  
+            },
+        }
+    }
+}
+
+
+// manually implement deserialize because Split is not a unit-struct but is
+// serialized like one.
+impl<'de> Deserialize<'de> for Split {
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserializer.deserialize_map(SplitVisitor)
+    }
+}
+
+struct SplitVisitor;
```

Comment:
I copied this functionality from the `Whitespace` pre-tokenizer. Not sure at all if this should be the same for `Split` to be honest.